File: core/src/com/alta_v2/rendering/component/animation/AnimationFactory.java
package com.alta_v2.rendering.component.animation;

import com.badlogic.gdx.math.Vector2;
import com.google.inject.assistedinject.Assisted;

public interface AnimationFactory {

    TranslationAnimation createFadeInAnimation(@Assisted(value = "start") Vector2 start,
                                               @Assisted(value = "target") Vector2 target,
                                               float duration);

    FadeAnimation creteFadeAnimation(@Assisted("startAlpha") float startAlpha,
                                     @Assisted("endAlpha") float endAlpha,
                                     @Assisted("duration") float duration);

}

File: core/src/com/alta_v2/rendering/common/component/animation/AnimationFactory.kt
package com.alta_v2.rendering.common.component.animation

import com.badlogic.gdx.math.Vector2
import com.google.inject.assistedinject.Assisted

interface AnimationFactory {

    fun createFadeInAnimation(@Assisted(value = "start") start: Vector2,
                              @Assisted(value = "target") target: Vector2,
                              duration: Float): TranslationAnimation

    fun creteFadeAnimation(@Assisted("startAlpha") startAlpha: Float,
                           @Assisted("endAlpha") endAlpha: Float,
                           @Assisted("duration") duration: Float): FadeAnimation
}